python/cudf/cudf/tests/test_column.py

```diff
@@ -136,3 +136,155 @@ def test_column_chunked_array_creation():
+        (np.arange(1), "int8", "datetime64[ns]"),
+    ],
+)
+def test_column_view_invalid_numeric_to_numeric(data, from_dtype, to_dtype):
+    cpu_data = np.asarray(data, dtype=from_dtype)
+    gpu_data = as_column(data, dtype=from_dtype)
+
+    try:
+        cpu_data = cpu_data.view(to_dtype)
+    except ValueError as error:
+        if "size must be a divisor" in str(error):
+            with pytest.raises(
+                TypeError, match="Can not divide",
+            ):
+                gpu_data = gpu_data.view(to_dtype)
+        else:
+            raise error
+
+
+@pytest.mark.parametrize(
+    "data,to_dtype",
+    [
+        (["a", "b", "c"], "int8"),
+        (["ab"], "int8"),
+        (["ab"], "int16"),
+        (["a", "ab", "a"], "int8"),
+        (['abcd', 'efgh'], 'float32'),
+        (['abcdefgh'], 'datetime64[ns]')
+    ],
+)
+def test_column_view_valid_string_to_numeric(data, to_dtype):
+    def str_host_view(list_of_str):
+        return np.concatenate(
+            [
+                np.frombuffer(s.encode("utf-8"), dtype=to_dtype)
+                for s in list_of_str
+            ]
+        )
+
+    expect = cudf.Series(cudf.Series(data)._column.view(to_dtype))
+    got = cudf.Series(str_host_view(data))
+
+    assert_eq(expect, got)
+
+
+def test_column_view_nulls_widths_even():
+
+    data = [1, 2, None, 4, None]
+    expect_data = [
+        np.int32(val).view("float32") if val is not None else np.nan
+        for val in data
+    ]
+
+    sr = cudf.Series(data, dtype="int32")
+    expect = cudf.Series(expect_data, dtype="float32")
+    got = cudf.Series(sr._column.view("float32"))
+
+    assert_eq(expect, got)
+
+    data = [None, 2.1, None, 5.3, 8.8]
+    expect_data = [
+        np.float64(val).view("int64") if val is not None else val
+        for val in data
+    ]
+
+    sr = cudf.Series(data, dtype="float64")
+    expect = cudf.Series(expect_data, dtype="int64")
+    got = cudf.Series(sr._column.view("int64"))
+
+    assert_eq(expect, got)
+
+
+def test_column_view_numeric_slice():
+
+    data = np.array([1, 2, 3, 4, 5], dtype="int32")
+    sr = cudf.Series(data)
+
+    expect = cudf.Series(data[1:].view("int64"))
+    got = cudf.Series(sr._column[1:].view("int64"))
+
+    assert_eq(expect, got)
+
+    expect = cudf.Series(data[1:3])
+    got = cudf.Series(sr._column[1:3].view("int64"))
+
+def test_column_view_string_slice():
+    def str_host_view(list_of_str, to_dtype):
+        return np.concatenate(
+            [
+                np.frombuffer(s.encode("utf-8"), dtype=to_dtype)
+                for s in list_of_str
+            ]
+        )
+    data = ['a', 'b', 'cd', 'efg', 'h']
+
+    expect = cudf.Series(cudf.Series(data)._column[3:].view('int8'))
+    got = cudf.Series(str_host_view(data[3:], 'int8'))
+    
+    assert_eq(expect, got)
+
+    expect = cudf.Series(cudf.Series(data)._column[2:5].view('int8'))
+    got = cudf.Series(str_host_view(data[2:5], 'int8'))
+
+    assert_eq(expect, got)
```

Comment:
new line at end of file.
Here is vscode settings to do it automatically,
```
"files.insertFinalNewline": true,
"files.trimFinalNewlines": true,
```

Reply:
Thanks! I didn't know you could do this automatically :)